### PR TITLE
Suggest hyperparam fixes when hyperparams don't validate against the schema

### PR DESCRIPTION
**Pull request by shinnar**
Addresses https://github.com/IBM/lale/issues/726

### CODE REVIEW

File: lale/operators.py [hunk 1]
```diff
@@ -1794,6 +1795,42 @@ def customize_schema(
...
+                    lale.type_checking.validate_schema(fixed_hp, hp_schema)
+                    found = True
+                    yield new_values
+                except jsonschema.ValidationError:
+                    pass
+
+    MAX_FIX_DEPTH: int = 2
+    MAX_FIX_SUGGESTIONS: int = 3
+
+    def _format_fix_suggestion(self, d: Dict[str, Any]) -> str:
```
Comment (hirzel) [inline lale/operators.py:1825]: I suggest to call lale.pretty_print.hyperparams_to_string here: https://github.com/IBM/lale/blob/master/lale/pretty_print.py#L80

File: lale/operators.py [hunk 2]
```diff
@@ -1794,6 +1795,42 @@ def customize_schema(
...
+        defaults = self.get_defaults()
+        explicit_defaults: Dict[str, Any] = {k: defaults[k] for k in key_candidates}
+
+        found: bool = False
+
+        for depth in range(0, max_depth):
+            if found:
+                return None
+            candidate_replacements: Any = list(
+                itertools.combinations(explicit_defaults.items(), depth + 1)
```
Comment (hirzel) [inline lale/operators.py:1810]: Nice use of itertools :)

**Review by hirzel — APPROVED** [review]
Thanks for this PR. Looks great :)

File: lale/operators.py (hunk 1 above)
Comment (shinnar) [reply lale/operators.py]: great idea, thanks!

**Pull request merged**